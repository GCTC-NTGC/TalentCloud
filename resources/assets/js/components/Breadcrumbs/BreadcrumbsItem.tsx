import * as React from "react";

interface BreadcrumbItemProps {
  href: string;
  title: string;
}

const BreadcrumbItem: React.FunctionComponent<BreadcrumbItemProps> = ({
  href,
  title,
}): React.ReactElement => {
  return (
    <li>
      <a href={href} title={title}>
        {title}
      </a>
      <i className="fa fa-caret-right" />
    </li>
  );
};

export default BreadcrumbItem;
