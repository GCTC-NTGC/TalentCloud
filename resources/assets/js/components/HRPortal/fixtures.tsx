import { text, boolean, number } from "@storybook/addon-knobs";
import { JobAction } from "../../models/types";
import { JobStatus } from "../../models/lookupConstants";

export const jobActions: JobAction[] = [
  {
    activity: {
      count: 3,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "CS01",
    draft: {
      url: "https://google.ca",
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: true,
    preview: {
      url: "https://google.ca",
      text: "Preview Poster",
      title: "Preview what this Job Poster will look like.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Draft,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Front-end Developer",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 12,
    classification: "BK01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Published,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Cupcake Baker",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "BS01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Complete,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Underwater Basket Weaver",
    userTime: 2,
  },
];

export const unclaimedJobs = [
  {
    title: "CS01 - Front-end Developer",
    createdAt: "2019-MAY-02",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: [],
  },
  {
    title: "AS02 - Executive Assisstant",
    createdAt: "2019-MAR-12",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: ["Rebecca Appleby", "Jack Little"],
  },
  {
    title: "ET03 - Business Analyst",
    createdAt: "2019-DEC-02",
    status: JobStatus.Review,
    hiringManagers: ["Robin Browne"],
    hrAdvisors: [],
  },
  {
    title: "FG05 - Long-term Care Specialist",
    createdAt: "2019-MAY-14",
    status: JobStatus.Draft,
    hiringManagers: ["Braeden McDoogal"],
    hrAdvisors: ["Caitlyn Summers", "Jack Little"],
  },
  {
    title: "CS03 - Digital Product Designer",
    createdAt: "2019-JUL-24",
    status: JobStatus.Review,
    hiringManagers: ["Amelie Lachance"],
    hrAdvisors: [],
  },
];
