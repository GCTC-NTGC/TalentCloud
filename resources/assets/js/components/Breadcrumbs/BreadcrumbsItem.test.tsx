import React from "react";
import { shallow } from "enzyme";
import { crumbs } from "./fixtures/crumbs";
import BreadcrumbsItem from "./BreadcrumbsItem";

describe("Breadcrumbs", (): void => {
  it("should render BreadcrumbsItem correctly passing crumbs prop", (): void => {
    const wrapper = shallow(<BreadcrumbsItem {...crumbs[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
