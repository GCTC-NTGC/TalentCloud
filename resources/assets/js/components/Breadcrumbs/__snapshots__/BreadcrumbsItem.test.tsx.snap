// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render BreadcrumbsItem correctly passing crumbs prop 1`] = `
<li>
  <a
    href="#"
    title="Manager Portal"
  >
    Manager Portal
  </a>
  <i
    className="fa fa-caret-right"
  />
</li>
`;
